### ISSUE TITLE
Codecs not working

### ISSUE THREAD

**Issue by riba2101** [mariadb-corporation/mariadb-connector-r2dbc#3]
Hi,

during testing/debugging I noticed a bug? 

org.mariadb.r2dbc.MariadbColumnMetadata#getJavaType

```
  @Override
  public Class<?> getJavaType() {
    return columnDefinitionPacket.getDataType().getDeclaringClass();
  }
```

always returns DataType, while 

org.mariadb.r2dbc.codec.list.LongCodec#canDecode

```
public boolean canDecode(ColumnDefinitionPacket column, Class<?> type) {
    return (COMPATIBLE_TYPES.contains(column.getDataType())
            || (column.getDataType() == DataType.BIGINT && column.isSigned()))
        && ((type.isPrimitive() && type == Integer.TYPE) || type.isAssignableFrom(Long.class));
  }
````

checks for `type.isAssignableFrom(Long.class)`

and this condition is always false, therefore the driver gives back
`Suppressed: io.r2dbc.spi.R2dbcTransientResourceException: [-1] [HY000] No decoder for type org.mariadb.r2dbc.codec.DataType and column type BIGINT(signed)`

This happens for DB types INTEGER and BIGINT, trying to map to Integer, Long or BigInteger.

shouldn't it be `return columnDefinitionPacket.getJavaType()`?

**Comment by rusher** [comment]
A lot of tests/corrections have been added to cover this area. 
Coverage is still not finished ( around 80% for now). 
This must have been corrected with this [commit](https://github.com/mariadb-corporation/mariadb-connector-r2dbc/commit/65c4e498a67f88b4613d87a2cbec5e74354dbe4f#diff-c3b92e3f7d07e9b9cdcc07cf34da94b4R50)

and will be released in the next version. 

(but available as snapshot using : )
```
<repositories>
    <!-- snapshot repository -->
    <repository>
        <id>sonatype-nexus-snapshots</id>
        <name>Sonatype Nexus Snapshots</name>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>

<dependencies>
    <dependency>
        <groupId>org.mariadb</groupId>
        <artifactId>r2dbc-mariadb</artifactId>
        <version>0.8.3-beta1-SNAPSHOT</version>
    </dependency>
</dependencies>
```
could you confirm resolution using snapshot ?

**Comment by rusher** [comment]
This is corrected in 0.8.3 beta version. Closing issue
